cudapoa/src/batch.cpp

```diff
@@ -18,25 +19,43 @@ namespace claragenomics
 namespace cudapoa
 {
 
-std::unique_ptr<Batch> create_batch(int32_t max_sequences_per_poa,
-                                    int32_t device_id,
+std::unique_ptr<Batch> create_batch(int32_t device_id,
                                     cudaStream_t stream,
                                     size_t max_mem,
                                     int8_t output_mask,
+                                    BatchSize batch_size,
```

Comment:
can we make `BatchSize` the first parameter? And take it in as a const ref? `const BatchSize&`

Reply:
done.

Reply:
I see the `const ref` change but not the BatchSize as the first param. I am little confused what we decided on now for it :)

Reply:
I thought it was agreed to keep device and stream ids as first parameters for now, since in many other constructors they come first. To keep consistency, I left the order as it is for ow and only changed the type to const ref. sorry for confusion, it looks we concluded our discussion with different takes!